core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -150,6 +150,7 @@ protected WritePrimaryResult shardOperationOnPrimary(BulkShardRequest request, I
                         final long version = indexResult.getVersion();
                         indexRequest.version(version);
                         indexRequest.versionType(indexRequest.versionType().versionTypeForReplicationAndRecovery());
+                        indexRequest.seqNo(indexResult.getSeqNo());
```

Comment:
maybe assert at the end of this method that the seqNo is set on the replica request?